README.md

```diff
@@ -1,72 +1,160 @@
-`/components/<component>/*/` - diversity.json from latest version of the component
+Response:
+["my-component2"]
+```
+#### Get diversity.json from component with version 1.0.0
+```
+GET /components/my-component2/1.0.0/
+
+Response:
+{
+  "name": "my-component2",
+  "title": "My Second Component",
+  "description": "A component that will do even more stuff",
+  "version": "1.0.0",
+  "angular": "myComponent2",
+  "template": "component.mustache.html",
+  "script": [
+    "scripts.min.js"
+  ],
+  "settings": {
+    "type": "object",
+    "properties": {
+      "classes": {
+        "type": "array",
+        "items": {
+          "type": "string"
+        }
+      }
+    }
+  },
+  "dependencies": {
+    "my-component1": "^0.0.1",
+  }
+}
+```
+#### Get diversity.json from latest patch-version of component 0.1.*
+```
+GET /components/my-component2/0.1/
+```
+#### Get diversity.json from latest version of the component
+```
+GET /components/my-component2/*/
+```
+#### All the component's CSS and SCSS compiled with the variables passed in via the query string
+```
+GET /components/my-component2/1.0.0/css?foo=bar
 
-`/components/<component>/1.2.3/css?foo=bar` - All the component's CSS and SCSS compiled with the variables passed in via the query string
-`/components/<component>/1.2.3/settings` - settings attribute from the diversity.json from component with version 1.2.3
-`/components/<component>/1.2.3/settingsForm` - settingsForm attribute from the diversity.json from component with version 1.2.3
-`/components/<component>/1.2.3/thumbnail` - thumbnail picutre from component with version 1.2.3
-`/components/<component>/1.2.3/files/path/to/file.js` - file.js from component with version 1.2.3
+Response: A CSS file
+```
+#### Get the settings attribute from the diversity.json from component with version 1.0.0
+```
+GET /components/my-components2/1.0.0/settings
+
+Response:
+{
+  "settings": {
+    "type": "object",
+    "properties": {
+      "classes": {
+        "type": "array",
+        "items": {
+          "type": "string"
+        }
+      }
+    }
+  }
+}
+```
+#### Get the settingsForm attribute from the diversity.json from component with version 1.2.3
+```
+GET /components/my-component/0.0.1/settingsForm
 
-Note: All the version selection methods can be used to get settings, settingsForm and files from the selected version.
+Response:
+{}
+```
+#### Get thumbnail picture from component with version 1.2.3
+```
+GET /components/<component>/1.2.3/thumbnail
 
+Response: Image data
+```
+#### Fetch file.js from component with version 1.0.0
+```
+GET /components/my-component2/1.0.0/files/script.min.js
 
-To register a component
------------------------
+Response: The requested files from the component
+```
 
-POST with parameter repo_url=url_to_repo
-`/components/<componentname>/register/`
+#### Note:
+All the version selection methods can be used to get settings, settingsForm and files from the selected version.
```

Comment:
I would love for it to say how to send the post data, if it's url-encoded, or form type, you know. Just a little note so we don't forget. Love the curl example below btw.